jest.setup.js

```diff
@@ -6,6 +6,24 @@ import 'jest-plugin-context/setup';
 import 'given2/setup';
 
 jest.mock('@/utils/rehypePrism');
+jest.mock('next/dynamic', () => ({
+  __esModule: true,
+  default: (...props) => {
+    const dynamicModule = jest.requireActual('next/dynamic');
+    const dynamicActualComp = dynamicModule.default;
+    const RequiredComponent = dynamicActualComp(props[0]);
+
+    if (RequiredComponent.preload) {
+      RequiredComponent.preload();
+
+      return RequiredComponent;
+    }
+
+    RequiredComponent.render.preload();
+
+    return RequiredComponent;
+  },
+}));
```

Comment:
https://stackoverflow.com/questions/46906948/how-to-unit-test-next-js-dynamic-components